pkg/minikube/download/download_test.go

```diff
@@ -197,3 +198,42 @@ func testImageToDaemon(t *testing.T) {
 		t.Errorf("Expected only 1 download attempt but got %v!", downloadNum)
 	}
 }
+
+// Validates that preload existence checks correctly caches values retrieved by remote checks.
+func testPreloadExistsCaching(t *testing.T) {
+	checkCache = func(file string) (fs.FileInfo, error) {
+		return nil, fmt.Errorf("cache not found")
+	}
+	doesPreloadExist := false
+	checkCalled := false
+	checkRemotePreloadExists = func(k8sVersion, containerRuntime string) bool {
+		checkCalled = true
+		setPreloadState(k8sVersion, containerRuntime, doesPreloadExist)
```

Comment:
It's kind of confusing why mocked `checkRemotePreloadExists` should update some state. Let's change PreloadExists to move this state change out of checkRemotePreloadExists. (see comment below)